### PR TITLE
Update SDK e2e test to reflect poetry changes

### PR DESCRIPTION
**Pull request by Srihari1192**
# Issue link

closes https://github.com/project-codeflare/codeflare-operator/issues/250

# What changes have been made

Updated the test `TestMNISTRayClusterSDK` to use poetry for installing CodeFlare SDK by replacing pip
Added the script `install-codeflare-sdk.sh` for installing codeflare-sdk using poetry and can reuse this in another tests

# Verification steps

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by sutaakar** [comment]
@Srihari1192 The `TestMNISTRayClusterSDK` is failing, can you check it?

**Comment by Srihari1192** [comment]
> @Srihari1192 The `TestMNISTRayClusterSDK` is failing, can you check it?

Test `TestMNISTRayClusterSDK` Failing at the step Waiting for requested resources to be set up cluster status as `queueing` and from operator logs seems the test failing because of `Insufficient resources to dispatch AppWrapper`..

```
    1 queuejob_controller_ex.go:1242] [ScheduleNext] [Agent Mode] Failed to dispatch app wrapper 'test-ns-qvqq7/mnist' due to insufficient resources, activeQ=true Unsched=false &qj=0xc0008a2c00 Version=5500 Status={Pending:0 Running:0 Succeeded:0 Failed:0 MinAvailable:0 CanRun:false IsDispatched:false State:Pending Message: SystemPriority:0 QueueJobState:HeadOfLine ControllerFirstTimestamp:2023-10-03 14:25:25.672931 +0000 UTC ControllerFirstDispatchTimestamp:0001-01-01 00:00:00 +0000 UTC FilterIgnore:true Sender:before ScheduleNext - setHOL Local:false Conditions:[{Type:Init Status:True LastUpdateMicroTime:2023-10-03 14:25:25.672931 +0000 UTC LastTransitionMicroTime:2023-10-03 14:25:25.672932 +0000 UTC Reason: Message:} {Type:Queueing Status:True LastUpdateMicroTime:2023-10-03 14:25:25.672945 +0000 UTC LastTransitionMicroTime:2023-10-03 14:25:25.672945 +0000 UTC Reason:AwaitingHeadOfLine Message:} {Type:HeadOfLine Status:True LastUpdateMicroTime:2023-10-03 14:25:25.695766 +0000 UTC LastTransitionMicroTime:2023-10-03 14:25:25.695766 +0000 UTC Reason:FrontOfQueue. Message:} {Type:Backoff Status:True LastUpdateMicroTime:2023-10-03 14:25:25.730656 +0000 UTC LastTransitionMicroTime:2023-10-03 14:25:25.730656 +0000 UTC Reason:AppWrapperNotRunnable. Message:Insufficient resources to dispatch AppWrapper.}] PendingPodConditions:[] TotalCPU:0 TotalMemory:0 TotalGPU:0 RequeueingTimeInSeconds:0 NumberOfRequeueings:0}```

**Comment by sutaakar** [comment]
@Srihari1192 That is because of resource limit for Ray head.
Can you try to apply head_cpus and head_mem with some low values?

**Review by Fiona-Waters — COMMENTED** [review]
I am running locally on a kind cluster and it is failing for me also. I have added 
`head_cpus=1,  head_memory=0.2,` to the cluster configuration but am still getting an error (I've tried increasing these also). The job is not reaching the desired `complete` or `failed` state. The end of the log looks like this...
image

**Comment by Srihari1192** [comment]
I think test were failing because of Ingress is not supporting in Kind cluster.. It should work after the [PR](https://github.com/project-codeflare/codeflare-sdk/pull/251)  merged

File: .github/actions/kind/action.yml [hunk 1]
```diff
@@ -48,6 +48,32 @@ runs:
         kubectl cluster-info
         kubectl describe nodes
 
+    - name: Create Ingress resource
```
Comment (astefanutti) [inline .github/actions/kind/action.yml:51]: I think it'd be more appropriate to have that Ingress created in the TestMNISTRayClusterSDK test itself.

File: test/e2e/install-codeflare-sdk.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# go to codeflare-sdk folder and install codeflare-sdk
+cd ..
+
+# Install Poetry and configure virtualenvs
+pip install poetry
+poetry config virtualenvs.create false
+
+cd codeflare-sdk
+# Clone the CodeFlare SDK repository
+git clone --branch main https://github.com/project-codeflare/codeflare-sdk.git
```
Comment (astefanutti) [inline test/e2e/install-codeflare-sdk.sh:12]: This should ideally be parameterized, but it'll probably be adapted in the context of project-codeflare/codeflare-sdk#306.

@sutaakar @ChristianZaccaria WDYT? Should we merge this PR as is and iterate on the SDK PR?

Comment (sutaakar) [reply test/e2e/install-codeflare-sdk.sh:12]: Fine for me.

Comment (ChristianZaccaria) [reply test/e2e/install-codeflare-sdk.sh:12]: /lgtm for merge.

**Review by ChristianZaccaria — APPROVED** [review]

**Pull request merged**